x-pack/test/security_solution_api_integration/test_suites/detections_response/utils/connectors/create_connector.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type SuperTest from 'supertest';
+
+interface CreateConnectorBody {
+  readonly name: string;
+  readonly config: Record<string, unknown>;
+  readonly connector_type_id: string;
+  readonly secrets: Record<string, unknown>;
+}
+
+export async function createConnector(
+  supertest: SuperTest.SuperTest<SuperTest.Test>,
+  connector: CreateConnectorBody,
+  id = ''
```

Comment:
Does the endpoint auto-create an `id` if it is not passed in the path?

Reply:
Yes, it's autogenerated. I planned to do connectors creation refactoring in a separate PR.